### DISCUSSION TITLE
PelePhysics data fitting of diffusion coefficients

### DISCUSSION THREAD

**Discussion by JayStandridge** [AMReX-Combustion/PeleC#607] [General]
Hello,

I am taking a look at how ddiag is calculated, and I was wondering if there was a paper I could read with how the coefficients stored in COFD (line 5185 in PelePhysics/Support/Mechanism/Models/LiDryer/mechanism.H in the PelePhysics section of the code) are generated. I did some looking around and couldn't find anything that used a data fit like it is done in the code, and the transport properties section in the PelePhysics documentation is empty.

**Comment by drummerdoc** [comment]
That comes originally from CHEMKIN's tranfit routines and is data that, originally in CHEMKIN was pre-computed and passed in the "linking files".  I'm pretty sure the gory details are in: 

Kee RJ, Dixon-Lewis G, Warnatz J, Coltrin ME, Miller JA. A FORTRAN Computer Code Package for the Evaluation of Gas-Phase Multicomponent Transport Properties. SANDIA; 1986.

**Reply by JayStandridge** [reply]
That is exactly what I needed, thank you so much!
Is there a place where I could look to find that type of information in the future?

**Reply by drummerdoc** [reply]
Gosh, I dunno.  I suppose there's a fair amount of history to all this.  The PeleC paper: 

Henry de Frahan MT, Rood JS, Day MS, et al. PeleC: An adaptive mesh refinement solver for compressible reacting flows. The International Journal of High Performance Computing Applications. 2022;0(0). doi:10.1177/10943420221121151

and original PeleLM(eX) (when it was LMC) paper:

M S Day & J B Bell (2000) Numerical simulation of laminar reacting flows with complex chemistry, Combustion Theory and Modelling, 4:4, 535-556, DOI: [10.1088/1364-7830/4/4/309](https://doi.org/10.1088/1364-7830/4/4/309)

Probably have much of the details you'd be looking for, but there are many angles here, including the AMReX bits, the combustion chemistry/transport, the numerical approaches, example science works.... hard to get 30 years of work in a single place.  Feel free to post more specific questions though and we'll do our best to help out.

**Reply by JayStandridge** [reply]
From what I can tell, it appears that PeleC does not compute the thermal diffusion coefficient or the effect of the thermal diffusion of species. It appears that the species/energy diffusion flux is calculated in Diffterm.H, where the correction velocity seems to be missing the thermal diffusion coefficient component. I have taken a look around the subroutines, and it doesn't look like this is accounted for elsewhere in the code. Is this correct?

**Comment by drummerdoc** [comment]
That is now part of the PeleLMeX code, but not yet in PeleC, though it wouldn't be hard to add, given that the infrastructure now exists to compute all the necessary bits.  (I assume, BTW, that you mean Soret, as opposed to conduction, which is definitely in there!)
